### PR TITLE
LOGGER.error(e.getMessage()) IS REALLY A BAD IDEA

### PR DESCRIPTION
**Pull request by liyuheng55555**

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/consensus/statemachine/dataregion/DataRegionStateMachine.java [hunk 1]
```diff
@@ -349,7 +349,7 @@ public DataSet read(IConsensusRequest request) {
       try {
         fragmentInstance = getFragmentInstance(request);
       } catch (IllegalArgumentException e) {
-        logger.error(e.getMessage());
+        logger.error(e.getMessage(), e);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/consensus/statemachine/dataregion/DataRegionStateMachine.java:352]: If use log.error("xxxx", e) will contains e.message, maybe we do not need to use `logger.error(e.getMessage(), e);`?

Comment (liyuheng55555) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/consensus/statemachine/dataregion/DataRegionStateMachine.java]: Indeed, but log4j interface always asks for a String parameter. There are some cases where I can only provide the exception stack and no additional information. In such cases, how should the string be written? Maybe just an empty string?

Comment (OneSizeFitsQuorum) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/consensus/statemachine/dataregion/DataRegionStateMachine.java]: Add some context for string?

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**